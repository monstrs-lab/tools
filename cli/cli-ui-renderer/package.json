{
  "name": "@monstrs/cli-ui-renderer",
  "version": "0.3.3",
  "license": "MIT",
  "type": "module",
  "exports": {
    "./package.json": "./package.json",
    ".": "./src/index.ts"
  },
  "main": "src/index.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "yarn library build",
    "prepack": "yarn run build",
    "postpack": "rm -rf dist"
  },
  "dependencies": {
    "ink": "^4.2.0",
    "react": "^18.2.0"
  },
  "devDependencies": {
    "@types/node": "^20.3.0",
    "@types/react": "^18.2.13"
  },
  "publishConfig": {
    "access": "public",
    "exports": {
      "./package.json": "./package.json",
      ".": {
        "import": "./dist/index.js",
        "types": "./dist/index.d.ts",
        "default": "./dist/index.js"
      }
    },
    "main": "dist/index.js",
    "types": "dist/index.d.ts"
  }
}
