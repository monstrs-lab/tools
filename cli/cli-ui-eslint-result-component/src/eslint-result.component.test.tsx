import React            from 'react'
import stripAnsi        from 'strip-ansi'

import { renderStatic } from '@monstrs/cli-ui-renderer'

import { ESLintResult } from './eslint-result.component.jsx'

describe('eslint result component', () => {
  it('render', () => {
    const value = {
      filePath: `${process.cwd()}/yarn/cli/src/tools/getPluginConfiguration.ts`,
      messages: [
        {
          ruleId: '@typescript-eslint/quotes',
          severity: 2,
          message: 'Strings must use singlequote.',
          line: 12,
          column: 44,
          nodeType: 'TemplateLiteral',
          messageId: 'wrongQuotes',
          endLine: 12,
          endColumn: 62,
          fix: { range: [447, 465], text: "'@yarnpkg/builder'" },
        },
      ],
      errorCount: 1,
      fatalErrorCount: 0,
      warningCount: 0,
      fixableErrorCount: 1,
      fixableWarningCount: 0,
      source:
        "/* eslint-disable import/no-dynamic-require */\n/* eslint-disable no-restricted-syntax */\n/* eslint-disable global-require */\n\nimport packageJson             from '@monstrs/yarn-cli/package.json'\nimport { PluginConfiguration } from '@yarnpkg/core'\n\nimport { getDynamicLibs }      from './getDynamicLibs'\n\nexport function getPluginConfiguration(): PluginConfiguration {\n  const plugins = new Set<string>()\n  for (const dependencyName of packageJson[`@yarnpkg/builder`].bundles.standard)\n    plugins.add(dependencyName)\n\n  const modules = getDynamicLibs()\n  for (const plugin of plugins) modules.set(plugin, require(plugin).default)\n\n  return { plugins, modules }\n}\n",
      usedDeprecatedRules: [
        {
          ruleId: 'lines-around-directive',
          replacedBy: ['padding-line-between-statements'],
        },
        { ruleId: 'global-require', replacedBy: [] },
        { ruleId: 'no-buffer-constructor', replacedBy: [] },
        { ruleId: 'no-new-require', replacedBy: [] },
        { ruleId: 'no-path-concat', replacedBy: [] },
      ],
    }

    const output = renderStatic(<ESLintResult {...value} />, 160)

    expect(stripAnsi(output)).toMatchSnapshot()
  })
})
