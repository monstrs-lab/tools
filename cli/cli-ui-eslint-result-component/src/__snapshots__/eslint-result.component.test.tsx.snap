// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eslint result component render 1`] = `
"yarn/cli/src/tools/getPluginConfiguration.ts:12:44

  Error: Strings must use singlequote. @typescript-eslint/quotes

  10 | export function getPluginConfiguration(): PluginConfiguration {
  11 |   const plugins = new Set<string>()
> 12 |   for (const dependencyName of packageJson[\`@yarnpkg/builder\`].bundles.standard)
     |                                            ^
  13 |     plugins.add(dependencyName)
  14 |
  15 |   const modules = getDynamicLibs()
"
`;
